### DISCUSSION TITLE
Interoperability TF Nov 23rd, 2023

### DISCUSSION THREAD

**Discussion by Pepe-Marquez** [FAIRmat-NFDI/fairmat-taxonomy#9] [Meeting's minutes]
### Purpose of Building the Materials Properties Taxonomy
- **Taxonomy Publication in an Interoperable Schema**: 
  - Discussion on the exportability of the schema.
  - Concerns were raised about versioning issues, especially if we wanted to be coupled to NOMAD versions. Luca Ghiringhelli suggested a slower approach to versioning. This was discussed assuming that the python schema would be able to export to OWL or something related.
- **Classification of the properties in the GUI**: Properties classified in sections. For example, _bandgap_ within _electronic properties_.
- **Exposing JSON-LD**: The feasibility of embedding JSON-LD within our HTML to foster discoverability (schema.org markup).
- **Reusable Base Sections**: 
  - The need for base sections in parsers and custom schemas that lead to unified results. See the following on base sections:

### Base Sections for the Properties should:
- **Automatic Copy to Results**: The classes should be capable of automatically normalizing data to their place in the results section.
- **Category Inclusion**: There should be a feature to categorize these classes.
- **Provenance Tracking**: A system for tracking the provenance of data through references in the results section.
- **Glossary Documentation**: The necessity for a glossary of classes, for ease of re-use in parsers and custom schemas.
- **Mapping to OWL**: Potential for mapping these classes to the OWL (Web Ontology Language).

### Todos
- **Luca Ghiringhelli** @LucaGhiringhelli: 
  - Tasked with organizing the definition revision in collaboration with PIs and domain experts.
- **Design of the Base Section - Sascha Klawohn** @Sideboard: 
  - Explore the possibility of adding an additional attribute in Metainfo.
  - Discussion on the location in the code:
    - NOMAD schema plugin structure.
      - Direct inclusion in the fairmat-taxonomy repository?

**Comment by hampusnasstrom** [comment]
> ### Base Sections for the Properties should:
> - **Automatic Copy to Results**: The classes should be capable of automatically normalizing data to their place in the results section.
> - **Category Inclusion**: There should be a feature to categorize these classes.
> - **Provenance Tracking**: A system for tracking the provenance of data through references in the results section.
> - **Glossary Documentation**: The necessity for a glossary of classes, for ease of re-use in parsers and custom schemas.
> - **Mapping to OWL**: Potential for mapping these classes to the OWL (Web Ontology Language).

Sounds like a very productive meeting. I like the idea of having sections referring to properties that can then be *composed* into our data sections. However, I think we need to be clear with the terminology here. I would not call these "base sections" as we do not intend to inherit from them. A better term would be something like "property sections". These should then ideally have a "PropertyEditQuantity" or something like that so that they don't all show up as sub sections in the ELN view.
